src/libclient/result/resultmodel.cpp

```diff
@@ -0,0 +1,238 @@
+#include "resultmodel.h"
+#include "resultscheduler.h"
+#include "../types.h"
+#include "task/scheduledefinition.h"
+#include "scheduler/scheduler.h"
+#include "client.h"
+
+#include <QPointer>
+#include <QUuid>
+#include <QHash>
+
+class ResultModel::Private : public QObject
+{
+    Q_OBJECT
+
+public:
+    Private(ResultModel *q)
+    : q(q)
+    {
+        scheduler = Client::instance()->scheduler();
+    }
+
+    ResultModel *q;
+
+    QPointer<ResultScheduler> resultScheduler;
+    QPointer<Scheduler> scheduler;
+
+    QHash<TaskId, int> identToResult;
+    ExtResultList results;
+
+public slots:
+    void resultAdded(const QVariantMap &result);
+    void resultModified(const QVariantMap &result);
+};
+
+void ResultModel::Private::resultAdded(const QVariantMap &result)
+{
+    int position = results.size();
+
+    q->beginInsertRows(QModelIndex(), position, position);
+    identToResult.insert(TaskId(result.value("task_id").toInt()), position);
+    results = resultScheduler->results();
+    q->endInsertRows();
+}
+
+void ResultModel::Private::resultModified(const QVariantMap &result)
+{
+    int pos = identToResult.value(TaskId(result.value("task_id").toInt()));
+    Q_ASSERT(pos != -1);
```

Comment:
Please use the logging facility and don't quit the program with an assertion (fix this also in the ReportModel).